src/modules/uuv_att_control/uuv_att_control.hpp

```diff
@@ -0,0 +1,192 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ *
+ * This module is a modification of the rover attitide control module and is designed for the
+ * TUHH hippocampus.
+ *
+ * All the acknowledgments and credits for the fw wing app are reported in those files.
+ *
+ * @author Daniel Duecker <daniel.duecker@tuhh.de>
+ * @author Philipp Hastedt <philipp.hastedt@tuhh.de>
+ * @author Tim Hansen <t.hansen@tuhh.de>
+ */
+
+
+
+#include <float.h>
+
+#include <drivers/drv_hrt.h>
+#include <lib/ecl/geo/geo.h>
+#include <lib/mathlib/mathlib.h>
+#include <lib/perf/perf_counter.h>
+#include <lib/pid/pid.h>
+#include <matrix/math.hpp>
+#include <px4_platform_common/px4_config.h>
+#include <px4_platform_common/defines.h>
+#include <px4_platform_common/posix.h>
+#include <px4_platform_common/tasks.h>
+#include <px4_platform_common/module.h>
+#include <px4_platform_common/module_params.h>
+#include <uORB/Subscription.hpp>
+#include <uORB/Publication.hpp>
+#include <uORB/topics/manual_control_setpoint.h>
+#include <uORB/topics/parameter_update.h>
+#include <uORB/topics/position_controller_status.h>
+#include <uORB/topics/sensor_combined.h>
+#include <uORB/topics/vehicle_acceleration.h>
+#include <uORB/topics/vehicle_attitude.h>
+#include <uORB/topics/vehicle_attitude_setpoint.h>
+#include <uORB/topics/vehicle_angular_velocity.h>
+#include <uORB/topics/vehicle_rates_setpoint.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/vehicle_local_position.h>
+#include <uORB/topics/actuator_controls.h>
+#include <uORB/topics/ekf2_timestamps.h>
+#include <uORB/uORB.h>
+
+
+
+
+using matrix::Eulerf;
+using matrix::Quatf;
+using matrix::Matrix3f;
+using matrix::Vector3f;
+using matrix::Dcmf;
+
+using uORB::SubscriptionData;
+
+class UUVAttitudeControl: public ModuleBase<UUVAttitudeControl>, public ModuleParams
+{
+public:
+	UUVAttitudeControl();
+	~UUVAttitudeControl();
+
+	UUVAttitudeControl(const UUVAttitudeControl &) = delete;
+	UUVAttitudeControl operator=(const UUVAttitudeControl &other) = delete;
+
+
+	/** @see ModuleBase */
+	static int task_spawn(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static UUVAttitudeControl *instantiate(int argc, char *argv[]);
+
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	/** @see ModuleBase::run() */
+	void run() override;
+
+
+//	int start();
+//	bool task_running() { return _task_running; }
+
+private:
+	uORB::Publication<position_controller_status_s>	_pos_ctrl_status_pub{ORB_ID(position_controller_status)};
+	uORB::Publication<actuator_controls_s>		    _actuator_controls_pub{ORB_ID(actuator_controls_0)};
+
+	uORB::Subscription	_parameter_update_sub{ORB_ID(parameter_update)};
+
+	int	_vehicle_attitude_sp_sub{-1};	/**< vehicle attitude setpoint */
```

Comment:
This is fine to merge if it's working for you, but you might want to update to the newer uORB c++ api at some point. 
Example https://github.com/PX4/Firmware/blob/cfa74b91063251b6d88ac7f65bbe1044ff67ffe6/src/modules/mc_rate_control/MulticopterRateControl.hpp#L96-L104

Reply:
Thanks!  I think for the moment that's fine but I will update this in an upcoming revision